app.py

```diff
@@ -125,22 +116,33 @@ def opt_out_of_experiments():
 @app.route(f"{URL_PREFIX}/unsubscribe")
 @auth.requires_login
 def unsubscribe():
-    account_id = auth.get_account_id()
-    with DB_ENGINE.connect() as conn:
-        account_repo = DbAccountRepository(conn)
-        account_repo.remove_subscription_for_account(account_id)
-        conn.commit()
-
-    return redirect(url_for("home", error_description="Sorry to see you go. You have been unsubscribed"))
+    return render_template("pre_unsubscribe.html")
 
 
-@app.route(f"{URL_PREFIX}/email_unsubscribe/<newsletter_id>")
-def email_unsubscribe(newsletter_id):
+@app.route(f"{URL_PREFIX}/pre_unsubscribe", methods=["POST"])
+@auth.requires_login
+def pre_unsubscribe():
+    main_menu_option = request.form.get("main-menu")
+    unsubscribe_menu_option = request.form.get("unsubscribe-menu")
+    error_description = "Please reach out to poprox admins to complete this request."
     with DB_ENGINE.connect() as conn:
         account_repo = DbAccountRepository(conn)
-        account_repo.remove_subscription_for_account(auth.get_account_id())
-        conn.commit()
-    return redirect(url_for("home", error_description="Sorry to see you go. You have been unsubscribed"))
+        if main_menu_option == "leave-experiment":
+            # opt-out
+            error_description = "this is an opt out request"
+        elif unsubscribe_menu_option == "remove-email":
+            account_repo.remove_subscription_for_account(auth.get_account_id())
+            account_repo.end_consent_for_account(auth.get_account_id())
+            error_description = "You have been unsubscribed from POPROX."
+        elif unsubscribe_menu_option == "withdraw-email-poprox":
+            account_repo.remove_email_for_account(auth.get_account_id())
+            error_description = "Your email has been withdrawn from POPROX."
+        elif unsubscribe_menu_option == "withdraw-all-data":
+            account_repo.set_deletion_for_account(auth.get_account_id())
```

Comment:
I think this would also implicitly end consent, and should probably nuke the email.

The way I've been understanding the choices is less:
"A, B, or C" and more "A, A+B, or A+B+C"

Reply:
I think I have fixed this for the new workflow